pkg/apis/apps/v1/defaults.go

```diff
@@ -84,13 +84,11 @@ func SetDefaults_DaemonSet(obj *appsv1.DaemonSet) {
 		}
 		if updateStrategy.RollingUpdate.MaxUnavailable == nil {
 			// Set default MaxUnavailable as 1 by default.
-			maxUnavailable := intstr.FromInt32(1)
-			updateStrategy.RollingUpdate.MaxUnavailable = &maxUnavailable
+			updateStrategy.RollingUpdate.MaxUnavailable = ptr.To(intstr.FromInt32(1))
```

Comment:
This doesn't really drop intstr.FromInt32?

Reply:
The description isn’t all that clear, I’ll fix it — the goal here wasn’t to remove uses of the `intstr` wrappers themselves, but to remove functions (and similar constructs) that were used to return pointers to `intstr` values. A more explicit example is https://github.com/kubernetes/kubernetes/pull/120008/files#diff-dba47173bb401012646c5e505c0fb434e7e2a84a92edfee8dbbed05efa27315dL530 which shows functions being removed.

While I was at it, I removed code which declared `intstr` variables only to take their address, as above, and cleaned up deprecated uses of `pointer` (since it ended up being weird having files importing `k8s.io/utils/pointer` and `k8s.io/utils/ptr`).